core/domain/question_domain.py

```diff
@@ -0,0 +1,215 @@
+# coding: utf-8
+#
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Domain objects relating to question."""
+
+from core.domain import exp_domain
+from core.platform import models
+from constants import constants
+import feconf
+import utils
+
+(question_models,) = models.Registry.import_models([models.NAMES.question])
+
+
+# Do not modify the values of these constants. This is to preserve backwards
+# compatibility with previous change dicts.
+QUESTION_PROPERTY_TITLE = 'title'
+QUESTION_PROPERTY_LANGUAGE_CODE = 'language_code'
+QUESTION_PROPERTY_QUESTION_DATA = 'question_data'
+
+# This takes additional 'property_name' and 'new_value' parameters and,
+# optionally, 'old_value'.
+CMD_UPDATE_QUESTION_PROPERTY = 'update_question_property'
+
+
+class QuestionChange(object):
+    """Domain object for changes made to question object."""
+    QUESTION_PROPERTIES = (
+        QUESTION_PROPERTY_TITLE, QUESTION_PROPERTY_QUESTION_DATA,
+        QUESTION_PROPERTY_LANGUAGE_CODE)
+
+    def __init__(self, change_dict):
+        """Initialize a QuestionChange object from a dict.
+
+        Args:
+            change_dict: dict. Represents a command. It should have a 'cmd'
+                key, and one or more other keys. The keys depend on what the
+                value for 'cmd' is. The possible values for 'cmd' are listed
+                below, together with the other keys in the dict:
```

Comment:
This docstring seems incomplete. I don't see the list...

Reply:
I am sorry. Can you please elaborate?

Reply:
Read the docstring. It seems to terminate midway. I don't see the list of possible values that is being referred to.

Reply:
Ahh! Updated.